README.md

```diff
@@ -483,6 +484,67 @@ b'\xca\x1ca7\xe6\xf5\xf4Kt\xe8'
 [7, 9]
 ```
 
+### Type Hints/Annotations
+
+- https://peps.python.org/pep-0483/
+- https://docs.python.org/3/library/typing.html
+- http://mypy-lang.org/
+- https://github.com/Microsoft/pyright
+
+Python is not a statically typed language. It is a strongly typed language as
+you cannot mix types at runtime:
+
+```python
+>>> 'a' + 5
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+TypeError: can only concatenate str (not "int") to str
+
+can only concatenate str (not "int") to str
+```
+
+unlike for example javascript:
+
+```js
+> 'a' + 5
+'a5'
+```
+
+However the strong runtime types dont really help with writing code in the editor
+as that can only utilise static types. Thats where python type hints come in.
```

Comment:
is it worth stipulating that this only applies to python3? hopefully everyone is using python3 anyway.